File: ac-cli/src/platforms/codex.ts
import { BasePlatform } from './base.js';
import type { CopyOptions } from '../types.js';

export class CodexPlatform extends BasePlatform {
  name = 'codex';
  configFile = 'AGENTS.md';
  skillsDir = '.codex/skills';
  userSkillsDir = '~/.codex/skills';
  subagentsDir = '';
  supportsSkills = false;
  supportsSubagents = false;

  async copyAgents(
    source: string,
    target: string,
    options: CopyOptions
  ): Promise<void> {
    await super.copyAgents(source, target, options);
  }

  async copySkill(
    _skillName: string,
    _sourceDir: string,
    _targetDir: string,
    _options: CopyOptions
  ): Promise<void> {
    console.log('Codex does not support skills. Skipping skill copy.');
  }

  async copySubagent(
    _agentName: string,
    _sourceDir: string,
    _targetDir: string,
    _options: CopyOptions
  ): Promise<void> {
    console.log('Codex does not support subagents. Skipping subagent copy.');
  }
}

File: ac-cli/internal/platform/codex.go
package platform

import (
	"fmt"

	"github.com/sebastiaanwouters/ac-cli/internal/config"
)

type CodexPlatform struct {
	*BasePlatform
}

func NewCodexPlatform() *CodexPlatform {
	return &CodexPlatform{
		BasePlatform: NewBasePlatform(
			"codex",
			"AGENTS.md",
			".codex/skills",
			"~/.codex/skills",
			"",
			false,
			false,
		),
	}
}

func (p *CodexPlatform) CopySkill(_, _, _ string, _ config.CopyOptions) error {
	fmt.Println("Codex does not support skills. Skipping skill copy.")
	return nil
}

func (p *CodexPlatform) CopySubagent(_, _, _ string, _ config.CopyOptions) error {
	fmt.Println("Codex does not support subagents. Skipping subagent copy.")
	return nil
}
